### PR TITLE
Kibana proxy

### PR DESCRIPTION
**Pull request by michalc**

### CODE REVIEW

File: core/app_outgoing.py [hunk 1]
```diff
@@ -84,63 +91,105 @@
...
+    we've blocked for > ttl and lost the lock. We _want_ to have more evidence
+    of this so we can address the problem.
+    '''
+    app_logger = logging.getLogger('activity-stream')
+    ttl = 2
+    aquire_interval = 1
+    extend_interval = 1
+    key = 'lock'
+
+    async def aquire():
```
Comment (lgarvey) [inline core/app_outgoing.py]: minor, bordering pedantic, but acquire has a 'c' in it.

File: core/app_outgoing.py [hunk 2]
```diff
@@ -84,63 +91,105 @@
...
 
+    indexes_without_alias, _ = await get_old_index_names(session, es_endpoint)
+    indexes_to_delete = indexes_matching_feeds(indexes_without_alias, [feed.unique_id])
+    await delete_indexes(session, es_endpoint, indexes_to_delete)
+
+    index_name = get_new_index_name(feed.unique_id)
+    await create_index(session, es_endpoint, index_name)
+    await create_mapping(session, es_endpoint, index_name)
+
     href = feed.seed
```
Comment (lgarvey) [inline core/app_outgoing.py]: I know this code isn't in the review, but I wonder if this code code be more self documenting.  What is a feed.seed? What is the href iterator?  Is it returning hrefs? Or _things_ from the href?

File: elasticsearch_proxy/app.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    token_path = '/o/token/'
+    me_path = '/api/v1/user/me/'
+    grant_type = 'authorization_code'
+    scope = 'read write'
+    response_type = 'code'
+
+    redirect_from_sso_path = '/__redirect_from_sso'
+    session_token_key = 'staff_sso_access_token'
+
+    def get_redirect_uri_1(session, request):
```
Comment (lgarvey) [inline elasticsearch_proxy/app.py]: Wonder if these could be slightly more descriptive

`get_redirect_uri_authenticate()` ?

File: elasticsearch_proxy/app.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        state = secrets.token_urlsafe(32)
+        set_redirect_uri_3(session, state, request)
+        redirect_uri_1 = urllib.parse.quote(get_redirect_uri_2(request), safe='')
```
Comment (lgarvey) [inline elasticsearch_proxy/app.py]: again - `redirect_uri_callback` ?

Comment (michalc) [reply elasticsearch_proxy/app.py]: `get_redirect_uri_1` -> `get_redirect_uri_authenticate`
`get_redirect_uri_2` -> `get_redirect_uri_callback`
`get_redirect_uri_3` -> `get_redirect_uri_final` ? (Not sure about this one... it's the one that that the user will finally end up after the whole thing...)

**Comment by lgarvey** [comment]
It's sort of like the django ?next= url

`get_redirect_uri_final` is descriptive enough

File: elasticsearch_proxy/app.py (hunk 4 above)
Comment (michalc) [reply elasticsearch_proxy/app.py]: Changed

File: core/app_outgoing.py (hunk 1 above)
Comment (michalc) [reply core/app_outgoing.py]: Changed

File: core/app_outgoing.py (hunk 2 above)
Comment (michalc) [reply core/app_outgoing.py]: > What is the href iterator? Is it returning hrefs? 

To answer this as well, `href` is the string href of the current page of activities being fetched. (There isn't really an iterator I think in the current version, it's just done with a `while` loop. It was tricky to add in the metrics I wanted using an interator)

Comment (michalc) [reply core/app_outgoing.py]: Happy ponder this / change, but for now will merge.

**Pull request merged**